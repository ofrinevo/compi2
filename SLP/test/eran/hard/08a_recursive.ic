	
class A {
	static int calc_print_factorial(int n) {
	 if (n <= 1) {
	  Library.printi(n);
	  Library.print(" ");
	  return n;
	 } else {
	  int a = A.calc_print_factorial(n-1);
	  a = a * n;
	  Library.printi(a);
	  Library.print(" ");
	  return a;
	 }
	}
} 

class Check {
	static void main (string[] args) { 
		
		A.calc_print_factorial(3);
		Library.println("");

		A.calc_print_factorial(5);
		Library.println("");
 	}
}
