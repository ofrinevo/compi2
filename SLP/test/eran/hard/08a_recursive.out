1 2 6 
1 2 6 24 120 
